tests/test-extensions/petstore-extension/src/main/java/org/mule/test/petstore/extension/PetStoreOperationsWithFailures.java

```diff
@@ -79,21 +83,76 @@ public Integer failOperationWithThrowable(@Connection PetStoreClient client) thr
         index++;
         return returnValue;
       }
+    };
+  }
+
+  @Throws(PetStoreCustomErrorProvider.class)
+  @OnException(PetStoreErrorHandler.class)
+  public PagingProvider<PetStoreClient, Integer> failPagedOperationWithErrorHandler(ErrorAction errorAction) {
+    return new AbstractPetStorePagingProvider(false) {
 
       @Override
-      public Optional<Integer> getTotalResults(PetStoreClient petStoreClient) {
-        return Optional.empty();
+      public List<Integer> getPage(PetStoreClient petStoreClient) {
+        throw new PetStoreException(errorAction);
       }
+    };
+  }
 
-      @Override
-      public void close(PetStoreClient petStoreClient) throws MuleException {
+  private abstract class AbstractPetStorePagingProvider implements PagingProvider<PetStoreClient, Integer> {
+
+    private boolean useStickyConnections;
+
+    public AbstractPetStorePagingProvider(boolean useStickyConnections) {
+      this.useStickyConnections = useStickyConnections;
+    }
+
+    @Override
+    public Optional<Integer> getTotalResults(PetStoreClient petStoreClient) {
+      return Optional.empty();
+    }
+
+    @Override
+    public void close(PetStoreClient petStoreClient) throws MuleException {
 
+    }
+
+    @Override
+    public boolean useStickyConnections() {
+      return useStickyConnections;
+    }
+
+  }
+
+  public static class PetStoreErrorHandler extends ExceptionHandler {
+
+    @Override
+    public Exception enrichException(Exception e) {
+      if (e instanceof PetStoreException) {
+        return ((PetStoreException) e).getException();
+      } else {
+        return e;
       }
+    }
+  }
 
-      @Override
-      public boolean useStickyConnections() {
-        return stickyConnections;
+  private static class PetStoreException extends RuntimeException {
+
+
+    private ErrorAction errorAction;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make this "errorAction" field final. ([squid:S1165](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1165))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7717&issues=AWkazdl7otIXnmsUFhSy&open=AWkazdl7otIXnmsUFhSy)

[ISSUE_KEY]: AWkazdl7otIXnmsUFhSy